### PR TITLE
Don't crash when tapping a device's hardware menu button

### PR DESCRIPTION
**Pull request by mkoch227**
In reference to #92 and #58, `onCreateOptionsMenu` in `MainActivity` has been updated:

 - Don't duplicate the notifications / search icons when tapping on the menu button (the purpose of the `hasInflated` tag is to prevent multiple creations)
 - Only register the notificationProvider if it isn't null (root cause of crashing on some devices, including mine :grinning:)

If there is a cleaner way to implement this fix, feel free to let me know!

### CODE REVIEW

**Comment by alorma** [comment]
Hi!

Thanks for this one!, in fact i fixed the duplicate menu entries, so, if you can update your PR with my changes it will be great! (take it from develop branch)

I will take a look after, and merge it.

**Comment by mkoch227** [comment]
I just pulled in the latest changes on the develop branch and removed my `hasInflated` check; however there is still a duplication.

On initial launch of app:
Imgur

After tapping the menu hardware key once:
Imgur

As you can see, the notification and search icons are duplicated when. However, the icons are not duplicated if I tap the hardware key again.

**Comment by alorma** [comment]
Uh_ ok :)

I will check it

**Comment by alorma** [comment]
OK LOL.

Can you update the PR? Or make a new one fixing it?

**Comment by mkoch227** [comment]
The original PR you merged won't duplicate items, so it should work as-is :).  I was just double-checking since you said you fixed the duplication issue, so I thought that my flag would be redundant.

**TL;DR** The original pull request takes care of it :smile:

**Comment by alorma** [comment]
Uhmmmm, so, in my branch (with hasInflated) it not happen? No duplicate?

I fixed that in another screen, sorry for the mesh :P

**Comment by mkoch227** [comment]
Correct. Your branch (with `hasInflated`) does not have the duplication issue

**Comment by alorma** [comment]
ok :)
